tests/components/brunt/test_config_flow.py

```diff
@@ -0,0 +1,186 @@
+    )
+    entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.brunt.config_flow.BruntClientAsync.async_login",
+        return_value=None,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}, data=CONFIG
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+@pytest.mark.parametrize(
+    "side_effect, error_message",
+    [
+        (ServerDisconnectedError, "cannot_connect"),
+        (ClientResponseError(None, None, status=403), "invalid_auth"),
+        (ClientResponseError(None, None, status=401), "unknown"),
+        (Exception, "unknown"),
+    ],
+)
+async def test_form_error(hass, side_effect, error_message):
+    """Test we handle cannot connect."""
+    with patch(
+        "homeassistant.components.brunt.config_flow.ClientResponseError.__str__",
+        return_value="ClientResponseError",
+    ), patch(
+        "homeassistant.components.brunt.config_flow.BruntClientAsync.async_login",
+        side_effect=side_effect,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}, data=CONFIG
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["errors"] == {"base": error_message}
+
+
+@pytest.mark.parametrize("side_effect", [None, Exception])
+async def test_reauth(hass, side_effect):
+    """Test uniqueness of username."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=CONFIG,
+        title="test-username",
+        unique_id="test-username",
+    )
+    entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.brunt.config_flow.BruntClientAsync.async_login",
+        side_effect=ClientResponseError(None, None, status=403),
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={
+                "source": config_entries.SOURCE_REAUTH,
+                "unique_id": entry.unique_id,
+                "entry_id": entry.entry_id,
+            },
+            data=None,
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "reauth_confirm"
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={},
+        )
+        assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result2["step_id"] == "reauth_input"
+    with patch(
+        "homeassistant.components.brunt.config_flow.BruntClientAsync.async_login",
+        return_value=None,
+        side_effect=side_effect,
+    ):
+        result3 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={"password": "test"},
+        )
+        if side_effect is None:
```

Comment:
Please don't use conditional logic in the tests. Add parameters to the test instead.